import "basic.proto";
import "user.proto";


//enum 
enum PBActivityType{
    PARTY = 1;
    TRAFFIC = 2;
    SHOPPING = 3;
}

enum PBActionType 
{
   ACTIVITY = 1;
   SHARE = 2;
   JOIN = 3;
}

enum PBActionStatus
{
    NORMAL = 1;
    DELETED = 2;
    MARK = 3; //Very important
}

message PBAlbum{
    required string album_id = 1;
    optional PBBriefUser user = 2;
    optional string name = 3;
    repeated string image_list = 4; 
}


enum PBPromotionStatus {
    RUNNING = 0;
    END = 1; 
    PENDDING = 2;
}

message PBPromotion
{
    required string promotion_id = 1;
    required int32 start_date = 2;
    required int32 end_date = 3;
    required string title = 4;
    required string content = 5;
    required PBPromotionStatus status = 6;
}

message PBMerchant
{
    required string merchant_id = 1;

//The attributes below defined are from aibang API
    optional string aibang_id = 2;
    optional string name = 3;
    optional string county = 4;
    optional string addr = 5;
    optional string tel = 6;
    optional string cate = 8;
    optional float rate = 9;
    optional int32 cost = 10;
    optional string desc = 11;
    optional float lng = 12;
    optional float lat = 13;
    optional string work_time = 14;
    optional string site_url = 15;
    optional string web_url = 16;
    optional string wap_url = 17;
    optional string img_url = 18;
    
    repeated PBPromotion promotions = 30;
}


enum PBPayType{
    AA = 1;
    FREE = 2;
    OTHER = 10;
}


message PBActivity{
    optional PBActivityType type = 1;
    optional string token = 2;
    optional PBContact content = 3;
    optional int32 join_deadtime = 4;
    optional int32 hold_deadtime = 5;
    optional PBPayType pay_type = 7;
    optional float budget = 8;
    optional float price = 9;
    optional int32 member_limit = 10;
    repeated string photo_list = 11;
    repeated PBBriefUser participants = 12;
    repeated PBBriefUser signups = 13;

    //count

    optional int32 comment_count = 30;
    optional int32 share_count = 31;
    optional int32 participant_count = 32;
    optional int32 mark_count = 33;
   
    //party && shopping attributes

    optional PBMerchant merchant = 50;
    optional PBLocation location = 51;


    //traffic attributes
    optional PBLocation start_loc = 70;
    optional PBLocation end_loc = 71;

}

message PBAction{
   required string action_id = 1;
   required PBActionType type = 2;
   required int32 c_date = 3;
   required PBBriefUser user = 4; //use for return

   optional PBActivity activity = 20; //used for create activity
   optional string comment = 21; //use for share or repost
   optional PBAction related_action = 22; //used for share or join
}



message PBComment {
    required string comment_id = 1;
    required PBBriefUser user  = 2;
    required int32 c_date = 3;
    required string action_id = 4;
    optional string content = 5;
    optional int32 star = 6;
    optional bool is_reply = 7;
    optional string reply_action_id = 8;
    optional string digest = 9;
}
